controllers/builder/openshiftbuilder.go

```diff
@@ -0,0 +1,273 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"context"
+	"strings"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imgv1 "github.com/openshift/api/image/v1"
+	buildclientv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kubeutil "github.com/kiegroup/kogito-serverless-operator/utils/kubernetes"
+
+	"github.com/kiegroup/kogito-serverless-operator/controllers/workflowdef"
+	"github.com/kiegroup/kogito-serverless-operator/utils"
+
+	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+)
+
+const (
+	imageStreamTagKind         = "ImageStreamTag"
+	defaultBuildMessageTrigger = "Triggered by Kogito Serverless Operator"
+)
+
+//		openshiftBuildPhaseMatrix Build phases correlations:
+//
+//	  - BuildPhaseScheduling: When we first schedule the build to the OpenShift cluster by creating a new BuildConfig for the workflow.
+//	    There's no Build instance at this point, if there's, it's in "New" phase.
+//	  - BuildPhasePending: When we fetch the Build, and we mimic its status, meaning that a pod has been scheduled to start the build.
+//	  - BuildPhaseRunning: --
+//	  - BuildPhaseSucceeded: "Complete" for the OCP Build
+//	  - BuildPhaseFailed: --
+//	  - BuildPhaseError: --
+//	  - BuildPhaseInterrupted: "Cancelled" for the OCP Build
+var openshiftBuildPhaseMatrix = map[buildv1.BuildPhase]operatorapi.BuildPhase{
+	buildv1.BuildPhaseNew:       operatorapi.BuildPhaseScheduling,
+	buildv1.BuildPhasePending:   operatorapi.BuildPhasePending,
+	buildv1.BuildPhaseRunning:   operatorapi.BuildPhaseRunning,
+	buildv1.BuildPhaseComplete:  operatorapi.BuildPhaseSucceeded,
+	buildv1.BuildPhaseFailed:    operatorapi.BuildPhaseFailed,
+	buildv1.BuildPhaseError:     operatorapi.BuildPhaseError,
+	buildv1.BuildPhaseCancelled: operatorapi.BuildPhaseInterrupted,
+}
+
+var _ BuildManager = &openshiftBuilderManager{}
+
+type openshiftBuilderManager struct {
+	buildManagerContext
+	buildClient buildclientv1.BuildV1Interface
+}
+
+func newOpenShiftBuilderManager(managerContext buildManagerContext, cliConfig *rest.Config) (BuildManager, error) {
+	manager := &openshiftBuilderManager{
+		buildManagerContext: managerContext,
+	}
+	buildClient, err := buildclientv1.NewForConfig(cliConfig)
+	if err != nil {
+		return nil, err
+	}
+	manager.buildClient = buildClient
+	return manager, err
```

Comment:
why not calling `newOpenShiftBuilderManagerWithClient` ?

Reply:
Because I created it after this code :P

I'll change.